### ISSUE TITLE
Build problem with recent versions of Nginx

### ISSUE THREAD

**Issue by berwynhoyt** [anaef/lws-nginx#2]
When built against recent versions of Nginx (e.g. latest stable 1.24.0), the following compile error results:
```
gcc -c -fPIC -pipe  -O -W -Wall -Wpointer-arith -Wno-unused-parameter -Werror -g  -I src/core -I src/event -I src/event/modules -I src/os/unix -I /usr/include/lua5.4 -I ../nginx-lws/src -I objs -I src/http -I src/http/modules \
	-o objs/addon/src/lws_lib.o \
	../nginx-lws/src/lws_lib.c
In file included from src/event/ngx_event.h:526,
                 from src/http/ngx_http_upstream.h:14,
                 from src/http/ngx_http.h:34,
                 from ../nginx-lws/src/lws_module.h:14,
                 from ../nginx-lws/src/lws_lib.h:13,
                 from ../nginx-lws/src/lws_lib.c:8:
src/event/ngx_event_udp.h:38:27: error: field ‘pkt6’ has incomplete type
   38 |     struct in6_pktinfo    pkt6;
      |                           ^~~~
```

This is because nginx-lws source files do not include the correct Nginx headers first, per [Nginx issue #2312](https://trac.nginx.org/nginx/ticket/2312). Including these at the start of every .c file fixes this problem, but you may not want them in quite every .c file, so I'll let you decide which ones require the patch.

**Comment by anaef** [comment]
Thanks for the detailed heads-up! I had not tested with NGINX beyond 1.18 yet, as LTS 22.04 is still using that version. I will resolve this later this week.

**Comment by anaef** [comment]
Resolved on main; this will go into release 0.9.1, together with the Lua 5.1/5.2 support.
